src/utils.ts

```diff
@@ -379,9 +379,9 @@ export const encodeBase58check = (value: Uint8Array): string => {
  * @returns The fingerprint of the public key.
  */
 export const getFingerprint = (publicKey: Uint8Array): number => {
```

Comment:
I'd consider rather refactoring this function to accept the expected key length from outside, so that it's more foolproof (the caller must be very well aware of the expected key lenght) as well as the error message. It looks a bit sketchy to me to report an error that the key must be of length x or y - it may imho just confuse whoever runs into it as usually crypto keys have a very precise length

this helper is used just in a few places so should be little effort to change

```suggestion
export const getFingerprint = (publicKey: Uint8Array, expectedKeyLength: 32 | 33): number => {
```